chapters/tutorials/index.rst

```diff
@@ -0,0 +1,403 @@
+
+We have everything ready to use our Flow, but so far we have been using dummy data, let's download our dataset now. We will use this snippet and we don't need to go into the details for this. What it does is to download the `covid dataset <https://www.kaggle.com/xhlulu/covidqa>`_.
+
+.. code-block:: python
+
+    def download_data(targets, download_proxy=None, task_name='download covid-dataset'):
+    """
+    Download data.
+
+    :param targets: target path for data.
+    :param download_proxy: download proxy (e.g. 'http', 'https')
+    :param task_name: name of the task
+    """
+    opener = urllib.request.build_opener()
+    opener.addheaders = [('User-agent', 'Mozilla/5.0')]
+    if download_proxy:
+        proxy = urllib.request.ProxyHandler(
+            {'http': download_proxy, 'https': download_proxy}
+        )
+        opener.add_handler(proxy)
+    urllib.request.install_opener(opener)
+    with ProgressBar(task_name=task_name, batch_unit='') as t:
+        for k, v in targets.items():
+            if not os.path.exists(v['filename']):
+                urllib.request.urlretrieve(
+                    v['url'], v['filename'], reporthook=lambda *x: t.update_tick(0.01)
+                )
+
+Let's re-organize our code a little bit. First, we should import everything we need:
+
+.. code-block:: python
+
+    import os
+    import urllib.request
+    import webbrowser
+    from pathlib import Path
+
+    from jina import Flow, Executor
+    from jina.logging import default_logger
+    from jina.logging.profile import ProgressBar
+    from jina.parsers.helloworld import set_hw_chatbot_parser
+    from jina.types.document.generators import from_csv
+
+Then we should have our `main`, a `donwload_data` function to get our data and a `tutorial` function for all the rest
+
+.. code-block:: python
+
+    def download_data(targets, download_proxy=None, task_name='download covid-dataset'):
+        #This is exactly as the previous snippet we just saw
+
+    def tutorial(args):
+        #Here we will have everything for our tutorial
+
+    if __name__ == '__main__':
+        args = set_hw_chatbot_parser().parse_args()
+        tutorial(args)
+
+Now let's see our `tutorial` function with all the code we've done so far:
+
+.. code-block:: python
+
+    def tutorial(args):
+        Path(args.workdir).mkdir(parents=True, exist_ok=True)
+
+        class MyTransformer(Executor):
+            def foo(self, **kwargs):
+                print(f'foo is doing cool stuff: {kwargs}')
+
+        class MyIndexer(Executor):
+            def bar(self, **kwargs):
+                print(f'bar is doing cool stuff: {kwargs}')
+
+        targets = {
+            'covid-csv': {
+                'url': args.index_data_url,
+                'filename': os.path.join(args.workdir, 'dataset.csv'),
+            }
+        }
+
+        # download the data
+        download_data(targets, args.download_proxy, task_name='download covid-dataset')
+
+        f = (
```

Comment:
- `f` -> `flow`

Reply:
Just checking: have we run `Black` on all our code blocks? That way they'll exactly mirror what we have in the real code in the repo